src/Elastic.Markdown/DocumentationGenerator.cs

```diff
@@ -99,6 +99,36 @@ public async Task ResolveDirectoryTree(Cancel ctx)
 		_logger.LogInformation("Resolving tree");
 		await DocumentationSet.Tree.Resolve(ctx);
 		_logger.LogInformation("Resolved tree");
+		ReportDuplicateTitles(DocumentationSet.Tree.ResolvedMarkdownFiles);
+	}
+
+	private void ReportDuplicateTitles(List<MarkdownFile> files)
+	{
+		// Create a dictionary where keys are the titles
+		// and values are files with that title
+		var titleMap = new Dictionary<string, List<MarkdownFile>>(StringComparer.OrdinalIgnoreCase);
```

Comment:
`HashSet<string>` will suffice we care about something being duplicate not necessary how many times since the diagnostics reporting will take care of that.

I think it makes more sense to implement this as `IMarkdownExporter` implementations (maybe `ValidationMarkdownExporter`). 

That way it works for both `docs-builder` and `docs-assembler`. Here the duplicate file checking is per `DocumentationSet` not overall.